provider/pkg/provider/util.go

```diff
@@ -133,3 +134,99 @@ func getActiveClusterFromConfig(config *clientapi.Config, overrides resource.Pro
 
 	return activeCluster
 }
+
+// pruneMap recursively drops keys from the source map that don't have a matching key in the target map and returns
```

Comment:
"drops keys from the source" is a misleading as the source isn't modified, is it?

Reply:
Good catch! That's a holdover from the initial implementation that I forgot to update.